diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index 7da572cbee48..91921e29fa7f 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -869,7 +869,7 @@
 			};
 
 			src: src@20d8000 {
-				compatible = "fsl,imx6sx-src", "fsl,imx51-src";
+				compatible = "fsl,imx6sx-src", "fsl,imx51-src", "syscon";
 				reg = <0x020d8000 0x4000>;
 				interrupts = <GIC_SPI 91 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 96 IRQ_TYPE_LEVEL_HIGH>;
