diff --git a/arch/arm/boot/dts/imx6sx-udoo-neo-full.dts b/arch/arm/boot/dts/imx6sx-udoo-neo-full.dts
index 96f4d89848a3..14ec58422181 100644
--- a/arch/arm/boot/dts/imx6sx-udoo-neo-full.dts
+++ b/arch/arm/boot/dts/imx6sx-udoo-neo-full.dts
@@ -15,6 +15,12 @@ memory@80000000 {
 		device_type = "memory";
 		reg = <0x80000000 0x40000000>;
 	};
+
+	imx6sx-cm4 {
+		compatible	= "fsl,imx6sx-cm4";
+		syscon		= <&src>;
+		clocks		= <&clks IMX6SX_CLK_M4>, <&clks IMX6SX_CLK_M4_PODF>;
+	};
 };
 
 &fec1 {
